lib/compiler.js

```diff
@@ -309,9 +310,17 @@ function resolveAppPath(rootDir, relativePath) {
   return resolvedPath;
 }
 
-function tryResolveAppPath(rootDir, relativePath) {
+function tryResolveAppPath(rootDir, relativePath, resolveOptions) {
+  /* In order to retain backward compatibility, while resolving
+   * component path, `resolveOptions` parameter is added where
+   * `resolveOptions.strict` = true,
+   * means retain backward compatibility when resolving the path and
+   * `resolveOptions.strict` = false,
+   * does not enforce any such restriction when resolving the path */
+  resolveOptions = resolveOptions || { strict: false };
+
   var fullPath = path.resolve(rootDir, relativePath);
-  if (fs.existsSync(fullPath))
+  if (fs.existsSync(fullPath) && !resolveOptions.strict)
```

Comment:
I think this is not correct, as you will reject absolute paths in strict mode. Perhaps the following implementation is more correct?

```js
 if (name[0] === '/') {
  fullPath = relativePath;
} else if (name.indexOf('./') === 0 || name.indexOf('../') === 0 || !resolveOptions.strict) {
  fullPath = path.resolve(rootDir, relativePath);
}

if (fullPath && fs.existsSync(fullPath)) {
  return fullPath;
}

// resolve as a module now
```